### ISSUE TITLE
Reg\Register an Operation\Page "Facility Information" - External user can register the operation

### ISSUE THREAD

**Issue by zoeyli-46** [bcgov/cas-registration#1895]
#### Description:
As an industry user, given that the operation has been added to my operator, I can register the operation for IRC approval.

#### Acceptance Criteria:
Given that I am on the Operation Information page of the New Registration form,
When I click on the save and continue button,
Then I am brought to the Facility Information page of the form
https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=11759-156&t=5omgZedIN5xnnFFl-4

~Given that I selected EIO as my registration purpose 
Then I do NOT see facility page, the facility page is skipped over
But, in the backend the system automatically creates a facility record for them attn: @andrea-williams~
pulled out into ticket #1962  

Given that I am on the Facility Information page of the New Registration form,
When I click Add Facility, 
Then it will display the add facility form. https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=12902-157&t=5omgZedIN5xnnFFl-4 

Given that I am on the Facility Information page of the New Registration form,
Then I see a table of all my facilities
If I have not added any facilities and the table is empty, it will have an empty state message: "No facilities to show, add a facility"

Given I clicked view details on a facility row
Then I can see the details of the facility 

#### Development Checklist:

- [ ] each step of the Registration workflow form hits a different API endpoint
- [ ] see notes added to wireframe (in red) for required vs. optional fields and conditionals associated with certain fields
- [ ] for Single Facility Operations (SFOs): facility name & type to be omitted from UI. A `Facility` record should still be created/updated in the database, with the following fields autopopulated (i.e., external user doesn't have to provide any input):
       - facility name = name of operation
       - facility type = 'Single Facility'
- [ ] for SFOs, the "Add Another Facility" button should not appear
- [ ] for Linear Facility Operations (LFOs), facility name and type to be included in UI 
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

**The facility DataGrid doesn't need to be done in this issue, the work for that is being done separately**

#### Dependencies

- Blocked by #1854 
- Blocking

**Comment by shon-button** [comment]
@patriciarussellCAS

Reachable at:
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/registration/register-an-operation

**Comment by patriciarussellCAS** [comment]
Review notes: 
- Works in Dev. Have not reviewed this in Test yet. 
- Will create a design backlog ticket for the Facility Information page to capture some suggestions to improve this flow for LFOs. 
- When I added a facility to an LFO operation there wasn't a continue button anywhere - where should users go next? Should they be sent to the next step in the process? This is also mentioned in #1608 
- If we use "back", "save and continue" buttons in this workflow, does it make sense to use it in the Admin module as well? 
- The SFO workflow currently sends users to the New Entrant Operation page, which is incorrect. But this is noted in other tickets.
